inspirehep/records/api/receivers.py

```diff
@@ -0,0 +1,44 @@
+import logging
+
+from flask_sqlalchemy import models_committed
+from invenio_records.models import RecordMetadata
+
+from inspirehep.pidstore.api import PidStoreBase
+from inspirehep.records.api import InspireRecord
+from inspirehep.records.errors import WrongOperationOnRecordError
+from inspirehep.records.indexer.tasks import index_record
+
+
+logger = logging.getLogger(__name__)
+
+
+@models_committed.connect
+def index_after_commit(sender, changes):
+    """Index a record in ES after it was committed to the DB.
+
+    This cannot happen in an ``after_record_commit`` receiver from Invenio-Records
+    because, despite the name, at that point we are not yet sure whether the record
+    has been really committed to the DB.
+    """
+    for model_instance, change in changes:
+        logger.debug("index_after_commit hook")
+        if isinstance(model_instance, RecordMetadata):
+            logger.debug(
```

Comment:
I would take this whole thing out and put it in a separate function/method, to leave only something like
```
if isinstance(model_instance, RecordMetadata):
    `after_record_metadata_commit(model_instance, change)`
```

Reply:
But this is already like a separate function. It's only doing this one thing.